files/scripts/update-cluster.sh

```diff
@@ -0,0 +1,137 @@
+  if grep -q 'HTTP/1.1 200' "clusterRemove.json"; then
+    echo "Scaling down successful."
+  else
+    echo "Issue scaling down:"
+    cat clusterRemove.json
+    echo
+    exit 1
+  fi
+}
+
+function addNodes {
+  local expectedNodes=$1
+  local authToken=$2
+  local namespace=$3
+  local timeout=$4
+  local currentNodes=$(getNodeCountInCurrentCluster "$authToken")
+  local nodes=""
+# if there is no cluster or current nodes are more or equal to expected so no need to add more, exit
+  if [ "$currentNodes" -lt 2 ] || [ "$currentNodes" -ge "$expectedNodes" ]; then
+    echo "No scaling required"
+    exit 0
+  fi
+  for ((i = currentNodes; i < expectedNodes; i++)) do
+    nodes=${nodes}\"graphdb-node-$i.graphdb-node.${namespace}.svc.cluster.local:7300\"
+    if [ $i -lt $(expr $expectedNodes - 1) ]; then
+      nodes=${nodes}\,
+    fi
+  done
+  nodes=\{\"nodes\":\[${nodes}\]\}
+  waitService "http://graphdb-cluster-proxy:7200/proxy/ready" "$authToken"
+  curl -o clusterAdd.json -isSL -m ${timeout} -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' --header "Authorization: Basic ${authToken}" -d "${nodes}"  'http://graphdb-cluster-proxy:7200/rest/cluster/config/node'
+  if grep -q 'HTTP/1.1 200' "clusterAdd.json"; then
+    echo "Scaling successful."
+  elif grep -q 'Mismatching fingerprints\|HTTP/1.1 412' "clusterAdd.json"; then
+    echo "Issue scaling:"
+    cat clusterAdd.json
+    echo
+    echo "Manual clear of the mismatched repositories will be required to add the node"
+    exit 0
+  else
+    echo "Issue scaling:"
+    cat clusterAdd.json
+    echo
+    exit 1
+  fi
+}
+
+function deleteCluster {
+  local authToken=$1
+  waitService "http://graphdb-node-0.graphdb-node:7200/rest/repositories" "$authToken"
+  curl -o response.json -isSL -m 15 -X DELETE --header "Authorization: Basic ${authToken}" --header 'Accept: */*' 'http://graphdb-node-0.graphdb-node:7200/rest/cluster/config?force=false'
+  if grep -q 'HTTP/1.1 200' "response.json"; then
+    echo "Cluster deletion successful!"
+  elif grep -q 'Node is not part of the cluster.\|HTTP/1.1 412' "response.json" ; then
+    echo "Node 0 is not part of a cluster"
+  else
+    echo "Cluster deletion failed, received response:"
+    cat response.json
+    echo
+    exit 1
+  fi
+}
+
+function getNodeCountInCurrentCluster {
+  local authToken=$1
+  local node_address=http://graphdb-node-0.graphdb-node:7200
+  waitService "${node_address}/rest/repositories" "$authToken"
+  curl -o clusterResponse.json -isSL -m 15 -X GET --header 'Content-Type: application/json' --header "Authorization: Basic ${authToken}" --header 'Accept: */*' "${node_address}/rest/cluster/config"
+  grep -o 'graphdb-node-' "clusterResponse.json" | grep -c ""
```

Comment:
this might fail if no cluster exists

Reply:
If no cluster exists the response will have zero instances of "graphdb-node-" so currentNode is set to 0 and everything is fine?